packages/database/src/core/SyncTree.ts

```diff
@@ -423,6 +423,38 @@ export function syncTreeRemoveEventRegistration(
   return cancelEvents;
 }
 
+/**
+ * This function was added to support non-listener queries,
+ * specifically for use in repoGetValue. It sets up all the same
+ * local cache data-structures (SyncPoint + View) that are
+ * needed for listeners without installing an event registration.
+ * If `query` is not `loadsAllData`, it will also provision a tag for
+ * the query so that query results can be merged into the sync
+ * tree using existing logic for tagged listener queries.
+ *
+ * @param syncTree - Synctree to add the query to.
+ * @param query - Query to register
+ * @returns tag as a string if query is not a default query, null if query is not.
+ */
+export function syncTreeRegisterQuery(syncTree: SyncTree, query: QueryContext) {
+  if (!query._queryParams.loadsAllData()) {
```

Comment:
I'm not sure if this is incorrect, but it's worth pointing out that the `syncTreeAddEventRegistration` for a default query will do the logic of L441 to L452.

I'm not sure why the [assertion](https://github.com/firebase/firebase-js-sdk/blob/cdada6c68f9740d13dd6674bcb658e28e68253b6/packages/database/src/core/SyncTree.ts#L857-L858) in `syncTreeApplyTaggedQueryOverwrite` doesn't fail in tests for default queries, shouldn't it be set in `syncTreeRegisterSyncPoint`?

Reply:
Will the tests still pass if you push this branch down to:
```
if (!query._queryParams.loadsAllData()) {
  return syncTreeTagForQuery_(syncTree, query);
} else {
  return null;
}
```

Reply:
Don't default queries not have tags?

Reply:
Corrected.